internal/component/prometheus/exporter/blackbox/blackbox.go

```diff
@@ -118,15 +164,42 @@ func (a *Arguments) Validate() error {
 		return fmt.Errorf("invalid blackbox_exporter config: %s", err)
 	}
 
+	if len(a.Targets) != 0 && len(a.TargetsList) != 0 {
```

Comment:
opposite to this validation scenario, is it invalid if both are length 0?

Reply:
yes because the targets list can be provided at runtime. It's a valid scenario to go from 0 targets to 2 targets and then back to 0.